Module not found: Error: Can't resolve 'jsonld'

Dear dhlab,
I got the following error after starting my app with ng s. 

```
ERROR in ./node_modules/@knora/core/fesm5/knora-core.js
Module not found: Error: Can't resolve 'jsonld' in '/Users/system/Desktop/nie-frontend/node_modules/@knora/core/fesm5'
```

I solved it with 

```
npm install jsonld
```


I installed knora - gui the following way:

```
yarn add @knora/core json2typescript@^1.0.5
```

environment.ts:
```
export const environment = {
  production: false,
  name: 'Salsah',
  api: 'http://0.0.0.0:3333',
  app: 'http://localhost:4200',
  media: 'http://localhost:1024'
};
```

app.module.ts
```
import {environment} from '../environments/environment';
import {KuiCoreConfig, KuiCoreModule} from '@knora/core';
..

        KuiCoreModule.forRoot({
            name: environment.name,
            api: environment.api,
            media: environment.media,
            app: environment.app
        }),
```

Would you like to include this package directly in the npm - module knora/core so that it will be installed automatically? Otherwise I can change the readme on npm to point out that the installation of this is required if it's ok for you.
Thank you very much,
Jan

--- a/projects/knora/authentication/package.json
+++ b/projects/knora/authentication/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@knora/authentication",
-  "version": "0.0.3",
+  "version": "0.0.4",
   "description": "Knora ui module",
   "repository": {
     "type": "git",
@@ -18,6 +18,6 @@
     "@angular/animations": "^6.0.0-rc.0 || ^6.0.0",
     "@angular/cdk": "^6.0.0-rc.0 || ^6.0.0",
     "@angular/material": "^6.0.0-rc.0 || ^6.0.0",
-    "@knora/core": "1.0.1-alpha"
+    "@knora/core": "^1.0.3-alpha"
   }
 }

--- a/projects/knora/core/README.md
+++ b/projects/knora/core/README.md
@@ -6,11 +6,11 @@ This module is part of [Knora-ui](https://github.com/dhlab-basel/Knora-ui) modul
 The core module contains every service to use Knora's RESTful webapi.
 
 ## Install
-`$ yarn add @knora/core json2typescript@^1.0.5`
+`$ yarn add @knora/core json2typescript@^1.0.5 jsonld`
 
 OR
 
-`$ npm install --save @knora/core json2typescript@^1.0.5`
+`$ npm install --save @knora/core json2typescript@^1.0.5 jsonld`
 
 
 ## Setup
@@ -32,6 +32,7 @@ export const environment = {
 };
 ```
 
+- name: Name of the app. We're using it as a title in the authentication module
 - api: set the url of the [Knora](https://www.knora.org) webapi server
 - app: on which url is this app running?
 - media: url of a specific media server. In our case it's [sipi](http://www.sipi.io)

--- a/projects/knora/core/README.md
+++ b/projects/knora/core/README.md
@@ -12,9 +12,15 @@ OR
 
 `$ npm install --save @knora/core json2typescript@^1.0.5 jsonld`
 
-
 ## Setup
-Import the core module in your app.module.ts 
+On version 6 of Angular CLI they removed the shim for global and other node built-ins as mentioned in [#9827 (comment)](https://github.com/angular/angular-cli/issues/9827#issuecomment-369578814). Because of the jsonld package, we have to manually shimming it inside of the polyfills.ts file of the app:
+```
+// Add global to window, assigning the value of window itself.
+
+ (window as any).global = window;
+```
+
+Next step is to import the core module in your app.module.ts 
 
 `import {KuiCoreConfig, KuiCoreModule} from '@knora/core';`
 

--- a/projects/knora/core/package.json
+++ b/projects/knora/core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@knora/core",
-  "version": "1.0.4-alpha",
+  "version": "1.0.5-alpha",
   "description": "Knora ui module",
   "repository": {
     "type": "git",

--- a/projects/knora/core/package.json
+++ b/projects/knora/core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@knora/core",
-  "version": "1.0.3-alpha",
+  "version": "1.0.4-alpha",
   "description": "Knora ui module",
   "repository": {
     "type": "git",
@@ -15,6 +15,7 @@
   "peerDependencies": {
     "@angular/common": "^6.0.0-rc.0 || ^6.0.0",
     "@angular/core": "^6.0.0-rc.0 || ^6.0.0",
-    "json2typescript": "^1.0.5"
+    "json2typescript": "^1.0.5",
+    "jsonld": "^1.0.2"
   }
 }

